src/main/java/com/hazelcast/aws/AWSClient.java

```diff
@@ -29,43 +28,43 @@
 public class AWSClient {
 
     private final AwsConfig awsConfig;
-
+    private final AwsClientStrategy clientStrategy;
     private String endpoint;
 
     public AWSClient(AwsConfig awsConfig) {
         if (awsConfig == null) {
             throw new IllegalArgumentException("AwsConfig is required!");
         }
         this.awsConfig = awsConfig;
-        this.endpoint = awsConfig.getHostHeader();
+        String hostHeader = awsConfig.getHostHeader();
+        this.endpoint = hostHeader;
         if (awsConfig.getRegion() != null && awsConfig.getRegion().length() > 0) {
-            if (!awsConfig.getHostHeader().startsWith("ec2.")) {
-                throw new InvalidConfigurationException("HostHeader should start with \"ec2.\" prefix");
+            if (!(hostHeader.startsWith("ec2.") || hostHeader.startsWith("ecs."))) {
+                throw new InvalidConfigurationException("HostHeader should start with \"ec2.\" or \"ecs.\" prefix, found: " + hostHeader);
             }
-            setEndpoint(awsConfig.getHostHeader().replace("ec2.", "ec2." + awsConfig.getRegion() + "."));
+            String host = hostHeader.substring(0, 4);
+            this.endpoint = hostHeader.replace(host, host + awsConfig.getRegion() + ".");
```

Comment:
For Chinese customers, AWS has a separate domain (`amazonaws.com.cn` IIRC) so it may be helpful to specify a custom hostHeader. But I agree we might make it dynamic somehow.

Reply:
@googlielmo, Could you add this dynamic behavior? You can create a GH Issue and send it as a separate PR if you prefer.

Reply:
Good point, I'll do it as a separate (presumably small) PR.

Reply:
opened #97